### PR TITLE
EditorProperty: Fix missing increment buttons for integers

### LINKED ISSUE #73192: 4.0.rc1, inc-decrement buttons missing from process priority input box
### Godot version

v4.0.rc1.official [8843d9ad3]

### System information

Windows 10 22H2, Compatibility, Geforce GTX 550 Ti

### Issue description

The inc-decrement buttons are missing from the Process Priority input box.

The issue is observed in: v4.0.rc1.official [8843d9ad3]

Godot_v4 0-rc1_win64_vzNcfx2YGm

The expected behavior is observed in: v3.5.2.rc2.official [1a2bf3eb4] and v3.5.1.stable.official [6fed1ffa3]

Godot_v3 5 2-rc2_win64_LYSNq0gjW1

### Steps to reproduce

- open the attached project: [issue, inc-decrement buttons missing from process priority input box.zip](https://github.com/godotengine/godot/files/10717722/issue.inc-decrement.buttons.missing.from.process.priority.input.box.zip)

- in the `Inspector` panel, left-click on the `Process` property group to unfold it's properties.
Look at the `Priority` input box.

### Expected
There should be increment and decrement buttons on the right side of the input box.

### Observed
There are no increment and decrement buttons on the right side of the input box.

### v4.0.rc1.official [8843d9ad3]

[issue, inc-decrement buttons missing from process priority input box.zip](https://github.com/godotengine/godot/files/10717722/issue.inc-decrement.buttons.missing.from.process.priority.input.box.zip)

### v3.5.2.rc2.official [1a2bf3eb4] and v3.5.1.stable.official [6fed1ffa3]

[inc-decrement buttons in process priority input box.zip](https://github.com/godotengine/godot/files/10717723/inc-decrement.buttons.in.process.priority.input.box.zip)

### PR DESCRIPTION
**Pull request by akien-mga**
Fixes #73192.

### CODE REVIEW

File: editor/editor_properties.cpp [hunk 1]
```diff
@@ -4174,9 +4174,12 @@ struct EditorPropertyRangeHint {
 	bool radians = false;
 };
 
-static EditorPropertyRangeHint _parse_range_hint(PropertyHint p_hint, const String &p_hint_text, double p_default_step) {
+static EditorPropertyRangeHint _parse_range_hint(PropertyHint p_hint, const String &p_hint_text, double p_default_step, bool is_int = false) {
 	EditorPropertyRangeHint hint;
 	hint.step = p_default_step;
+	if (is_int) {
+		hint.hide_slider = false; // Always show slider for ints, unless specified in hint range.
```
Comment (TokageItLab) [inline editor/editor_properties.cpp:4181]: There is no logic in the spin box to properly time the undo/redo as was done in #69797.

I remember there was a problem with this, that every time a button is pressed in FPS mode, an undo redo is recorded. There are places where the buttons should be hidden, so if you want them to show by default, we need to add a negative hint like `hidebutton`.

**Review by TokageItLab — CHANGES_REQUESTED** [review]

File: editor/editor_properties.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/editor_properties.cpp:4181]: AFAIK this PR still respects the `hide_slider` hint, which will hide the buttons.

Comment (TokageItLab) [reply editor/editor_properties.cpp:4181]: Ah I see, I will test it.

**Review by TokageItLab — APPROVED** [review]
I tested and it seems fine.

**Pull request merged**